misc/config_tools/configurator/packages/configurator/src/pages/Config.vue

```diff
@@ -211,7 +211,14 @@ export default {
         }
       }
       if (this.currentFormData['load_order'] === 'SERVICE_VM') {
-        delete this.currentFormSchema.BasicConfigType.properties.vm_type
+        if (this.currentFormData.hasOwnProperty('vm_type')) {
+          delete this.schemas.ServiceVM.BasicConfigType.properties.vm_type
```

Comment:
@junjiemao1 Was this the right way to remove the visibility for an option? Shouldn't this be done in the .xsd file so the xml schema can be validated?

Reply:
Hardcoded logic in the configurator is always not preferred. However, the current XSD attributes do not provide a way to hide a config item conditionally (which is quite complex as conditional things are always so).

I would consider this as a temporary fix to the issue we see. A right fix may involve further discussions about the config items and ways we want to show to users, which can hardly catch 3.0.